### PR TITLE
Prevent exporting consul service to peers.

### PR DESCRIPTION
**Pull request by hashi-derek**
Prior to this change, it was possible for the "consul" service to be exported. This change prevents it from being exported via wildcards, explicit exports, and discovery chains (resolvers, splitters, routers).

### CODE REVIEW

File: agent/consul/state/peering.go [hunk 1]
```diff
@@ -852,18 +865,24 @@ func exportedServicesForPeerTxn(
...
-			if err != nil {
-				return fmt.Errorf("failed to get discovery chain targets for service %q: %w", svc, err)
-			}
+		idx, targets, err := discoveryChainOriginalTargetsTxn(tx, ws, dc, svc.Name, &svc.EnterpriseMeta)
+		if err != nil {
+			return fmt.Errorf("failed to get discovery chain targets for service %q: %w", svc, err)
+		}
+		if idx > maxIdx {
+			maxIdx = idx
+		}
```
Comment (DanStough) [inline agent/consul/state/peering.go:874]: Does moving this out of the check for L4 protocols change the semantics of when we update the maxIdx? Seems like we might want to wait to increment until we decide if the targets are valid.

**Review by DanStough — COMMENTED** [review]
I just have one question.

File: agent/consul/state/peering.go (hunk 1 above)
Comment (hashi-derek) [reply agent/consul/state/peering.go:874]: Ah, I see what you mean. So maybe move the `if idx > maxIdx{` until right before the return for the function?

**Review by DanStough — APPROVED** [review]
LGTM 🚀

**Pull request merged**